### PR TITLE
Created api endpoint for creating post with tests

### PR DESCRIPTION
**Pull request by Chenxi96**
…cated, prisma migrate prisma schema, edit docker compose due to deprecation in package json

renamed mock next auth and created test for api endpoint

fixed typo issues

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have added thorough tests. (If tests are not applicable, explain why in the next section.)
- [x] Is there only 1 commit.

## Describe your changes
1. Fixed minor docker compose issues (deprecated version issue).
2. Updated Post model in Prisma.
3. Renamed mock file for next auth to next auth react.
4. Created API endpoint to create post record after authentication is checked.
5. Tested API endpoint.

### CODE REVIEW

File: __tests__/api/post/route.test.jsx
Comment (Chenxi96) [inline __tests__/api/post/route.test.jsx:1]: Created Test for checking API route endpoint

File: app/api/post/route.ts
Comment (Chenxi96) [inline app/api/post/route.ts:1]: Created API Route endpoint

File: jest.config.ts [hunk 1]
```diff
@@ -11,8 +11,8 @@ const config: Config = {
   setupFilesAfterEnv: ["<rootDir>/lib/prisma.ts"],
   testEnvironment: "jsdom",
   moduleNameMapper: {
-    'next-auth/react': '<rootDir>/mocks/auth/next-auth.ts',
-  }
+    'next-auth/react': '<rootDir>/mocks/auth/next-auth-react.ts',
```
Comment (Chenxi96) [inline jest.config.ts:14]: renamed file to match with module

File: lib/createPost.ts
Comment (Chenxi96) [inline lib/createPost.ts:1]: Created file for createPost method due to Nextjs Route error issue. Link: https://stackoverflow.com/questions/76214501/nextjs-13-folder-structure-best-practice

File: package.json [hunk 2]
```diff
@@ -8,8 +8,8 @@
     "start": "next start",
     "test": "NODE_ENV=test && jest --watch",
     "test:ci": "jest --ci --passWithNoTests",
-    "local": "docker-compose up -d && npx prisma studio",
-    "stop-local": "docker-compose down",
+    "local": "docker compose up -d && npx prisma studio",
```
Comment (Chenxi96) [inline package.json:11]: Needed to change due to docker version update. Link: https://stackoverflow.com/questions/78769459/command-not-found-docker-compose-after-docker-4-32-0-update

File: prisma/migrations/20240904235541_typo_fix/migration.sql
Comment (Chenxi96) [inline prisma/migrations/20240904235541_typo_fix/migration.sql:1]: Fixed typo

File: prisma/schema.prisma [hunk 3]
```diff
@@ -17,6 +17,8 @@ model Post {
   location     Unsupported("geography(Point, 4326)")?
   owner        User                                  @relation(fields: [ownerId], references: [id])
   ownerId      String                                @unique
+  description  String
```
Comment (Chenxi96) [inline prisma/schema.prisma:20]: Added fields to Prisma Post model

File: app/api/post/route.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+            title: data.title,
+            latitude: data.latitude,
+            longitude: data.longitude,
+            ownerId: data.ownerId,
+            description: data.description,
+            availableSpots: data.availabeSpots,
+            name: data.name,
+            email: data.email
+        })
+        return NextResponse.json({ message: 'Authenticated'}, {status: 201})
```
Comment (alvinc21) [inline app/api/post/route.ts]: I think 'Authenticated' as a response message is misleading for an endpoint that creates a post. Return something like 'Created'.

File: app/api/post/route.ts
Comment (alvinc21) [reply app/api/post/route.ts:1]: change the file path and route to `/api/posts`. It matches REST conventions to have a plural noun for a resource.

File: docker-compose.yml [hunk 5]
```diff
@@ -1,4 +1,3 @@
-version: '3.6'
```
Comment (alvinc21) [inline docker-compose.yml:1]: why was this deleted?

**Review by alvinc21 — COMMENTED** [review]
looking good

File: docker-compose.yml (hunk 5 above)
Comment (Chenxi96) [reply docker-compose.yml:1]: After updating docker, I started getting a warning. Apparently, with the new docker update the version property is obsolete. When running docker compose as default it will always use the newest version. link: https://forums.docker.com/t/docker-compose-yml-version-is-obsolete/141313/2

File: __tests__/api/post/route.test.jsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
+/**
+ * @jest-environment node
+ */
+
+import { POST } from '@/app/api/posts/route'
```
Comment (Chenxi96) [inline __tests__/api/post/route.test.jsx]: updated file path

Comment (alvinc21) [reply __tests__/api/post/route.test.jsx]: update the file path of the test too

**Review by alvinc21 — APPROVED** [review]
